### PR TITLE
Added Shader Library tests.

### PR DESCRIPTION
**Pull request by jasomobica**
Ported gles2 and gles3 Shader Library tests from Android deqp.

Signed-off-by: Janusz Sobczak <janusz.sobczak@mobica.com>

### CODE REVIEW

File: sdk/tests/deqp/resources/deqp-draw.js [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    TRIANGLE_STRIP: 1,
+    TRIANGLE_FAN: 2,
+
+    LINES: 3,
+    LINE_STRIP: 4,
+    LINE_LOOP: 5,
+
+    POINTS: 6,
+
+    PATCHES: 7
```
Comment (kenrussell) [inline sdk/tests/deqp/resources/deqp-draw.js]: The PATCHES primitive type doesn't exist in OpenGL ES 2.0 or WebGL. Should be removed so the code below throws an exception if somebody tries to use it.

File: sdk/tests/deqp/resources/shader-library-case.js [hunk 2]
```diff
@@ -0,0 +1,1144 @@
...
+};
+
+/**
+ * Initialize a test case
+ */
+var init = function() {
+    var state = deqpTests.runner.getState();
+    var test = state.currentTest;
+
+    _bufferedLogToConsole('Processing ' + test.fullName());
```
Comment (kenrussell) [inline sdk/tests/deqp/resources/shader-library-case.js]: This is "exported" as bufferedLogToConsole -- please use the public name.

File: sdk/tests/deqp/resources/shader-library-case.js [hunk 3]
```diff
@@ -0,0 +1,1144 @@
...
+    {
+        case expectResult.EXPECT_PASS:
+        case expectResult.EXPECT_VALIDATION_FAIL:
+            if (!allCompilesOk)
+                failReason = 'expected shaders to compile and link properly, but failed to compile.';
+            else if (!allLinksOk)
+                failReason = 'expected shaders to compile and link properly, but failed to link.';
+            break;
+
+        case expectResult.EXPECT_COMPILE_FAIL:
```
Comment (kenrussell) [inline sdk/tests/deqp/resources/shader-library-case.js]: It's not possible in OpenGL ES to expect compiles to fail; it's always legal for the shader compiler to defer compiles until link time. How do the tests use these expectations? I'd assume the negative tests should always use EXPECT_COMPILE_LINK_FAIL.

File: sdk/tests/resources/js-test-pre.js [hunk 4]
```diff
@@ -179,6 +190,57 @@ function testFailed(msg)
     _flushBufferedLogsToConsole();
 }
 
+/**
+ * Sets the current test name for usage within testPassedOptions/testFailedOptions.
+ * @param {string} name The name to set as the current test name.
+ */
+function setCurrentTestName(name)
+{
+	this._currentTestName = name;
```
Comment (kenrussell) [inline sdk/tests/resources/js-test-pre.js]: You can't rely on having a valid "this" pointer here. Just define _currentTestName in the global scope in this file and assign to it, and remove the definition in the anonymous function above.

File: sdk/tests/resources/js-test-pre.js [hunk 5]
```diff
@@ -179,6 +190,57 @@ function testFailed(msg)
...
+}
+
+/**
+ * Gets the current test name in use within testPassedOptions/testFailedOptions.
+ * @return {string} The name of the current test.
+ */
+function getCurrentTestName()
+{
+	return this._currentTestName;
```
Comment (kenrussell) [inline sdk/tests/resources/js-test-pre.js]: Remove "this." per above comment.

File: sdk/tests/00_test_list.txt [hunk 6]
```diff
@@ -3,5 +3,7 @@
 
 conformance/00_test_list.txt
 conformance/more/00_test_list.txt
+deqp/data/gles2/shaders/00_test_list.txt
+deqp/data/gles3/shaders/00_test_list.txt
```
Comment (kenrussell) [inline sdk/tests/00_test_list.txt]: Add "--min-version 2.0.0" at the beginning of this line, which will restrict the GLES3 shaders to only run as part of the WebGL 2.0 conformance suite.

File: sdk/tests/deqp/resources/deqp-draw.js (hunk 1 above)
Comment (kenrussell) [inline sdk/tests/deqp/resources/deqp-draw.js]: Actually, you can ignore this comment. It's fine to leave this in place for the time being.

**Comment by kenrussell** [comment]
Thank you; this is excellent work. Let me know whether you're willing to update this pull request or whether you'd rather merge it as is and submit new pull requests for the minor comments above. I'm happy to merge as is because the review is somewhat difficult with so many new files.

**Comment by jasomobica** [comment]
Let's merge as is. We can fix the issues later, while working on next tests.

Janusz
28 sty 2015 19:57 "Ken Russell" <notifications@github.com> napisał(a):

**Comment by kenrussell** [comment]
Sounds good. Merging.

**Pull request merged**